File: Python_solution/WordCount.py
from pyspark import SparkContext

# Initialize SparkContext
sc = SparkContext("local", "Word Count")

# Read the input file
text_file1 = sc.textFile("../input/sample1.txt")
text_file2 = sc.textFile("../input/sample2.txt")

# Perform word count
counts1 = (text_file1
          .flatMap(lambda line: line.split())
          .map(lambda word: (word, 1))
          .reduceByKey(lambda a, b: a + b))

# Perform word count
counts2 = (text_file2
          .flatMap(lambda line: line.split())
          .map(lambda word: (word, 1))
          .reduceByKey(lambda a, b: a + b))

# Save the output
counts1.saveAsTextFile("output1")
counts2.saveAsTextFile("output2")

sc.stop()

File: Java_solution/WordCount.java
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;

import java.util.Arrays;

public class WordCount {

    public static void main(String[] args) {
        // Set up Spark configuration and context
        SparkConf conf = new SparkConf().setAppName("Word Count").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        // Load the input files
        JavaRDD<String> textFile1 = sc.textFile("../input/sample1.txt");
        JavaRDD<String> textFile2 = sc.textFile("../input/sample2.txt");

        // Perform word count on the first file
        JavaPairRDD<String, Integer> counts1 = textFile1
                .flatMap(line -> Arrays.asList(line.split(" ")).iterator())
                .mapToPair(word -> new Tuple2<>(word, 1))
                .reduceByKey(Integer::sum);

        // Perform word count on the second file
        JavaPairRDD<String, Integer> counts2 = textFile2
                .flatMap(line -> Arrays.asList(line.split(" ")).iterator())
                .mapToPair(word -> new Tuple2<>(word, 1))
                .reduceByKey(Integer::sum);

        // Save the output for both word counts
        counts1.saveAsTextFile("output1");
        counts2.saveAsTextFile("output2");

        // Stop the Spark context
        sc.stop();
    }
}